packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -315,6 +315,10 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     const { iconProps } = this.props;
 
     if (iconProps) {
+      if (iconProps.className) {
+        const { className, ...rest } = iconProps;
```

Comment:
I actually don't think this conditional needs to be there - destructuring should work even if className doesn't exist. Then you can probably do css(...) like you do already...